src/Middleware/HttpOverrides/src/CertificateForwarderOptions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNetCore.HttpOverrides
+{
+    /// <summary>
+    /// Used to configure the <see cref="CertificateForwarderMiddleware"/>.
+    /// </summary>
+    public class CertificateForwarderOptions
+    {
+        /// <summary>
+        /// The name of the header containing the client certificate.
+        /// </summary>
+        /// <remarks>
+        /// This defaults to X-ARR-ClientCert, which is the header Azure Web Apps uses.
+        /// </remarks>
+        public string CertificateHeader { get; set; } = "X-ARR-ClientCert";
```

Comment:
Can we avoid this coupling in our stack? Default it to null and we can do the Azure specific cert options elsewhere (much like we do forwarded headers).

Reply:
Note defaulting to null would mean the config is invalid by default.

I don't see a defacto standard name for this one, "X-Client-Cert" is the closest I can find.

Reply:
Defaulting to something azure specific is a no go, so invalid by default means it should throw or 400 like we spoke about.

Reply:
For null it would throw on startup when Options validation ran.

Reply:
We wouldn't need the validation then, it could just no-op by default if the header is null/empty like it did before

Reply:
No, that's a pit of failure. If you're adding the middleware then you want it to work or tell you why not.

Reply:
So how do we want this to behave by default?  `UseCertificateHeaderForwarding` without calling Configure and specifying a header will throw and say please call `AddCertificateHeaderForwarding` and specify a header name?